### DISCUSSION TITLE
[Logic Apps] Cost Model

### DISCUSSION THREAD

**Discussion by jakubigla** [AzureCommunityPL/group#69] [Pomoc] [answered]
Czesc, staram sie porownac koszta implementacji "serverless workflow" miedzy AWS Step function i Azure Logic Apps.

W przypadku AWS wszystko jest jasne. Placisz za "zmiane stanu" $0.000025 + za to co "kazdy krok" wywoluje. Np za Lambde. 

W przypadku Azure, tez widze tak sama cene ($0.000025), za "Action", czyli podobna sprawa. Natomiast widze jakies dodatkowe oplaty za tzw "konektory". I z tego co wyczytalem, zeby wywolac wczesniej stworzona Azure function, to trzeba to zrobic przez konektor ktory dodatkowo kosztuje $0.000125 za wywolanie. A jeszcze oczywiscie do oplaty jest wywolanie samego Azure function. 

I teraz nie wiem, czy cos zle zrozumialem, ale wyglada na to ze jesli twoj workflow wywoluje jakies funkcje, to Azure jest kompletnie nie oplacalny?

**Comment by wi5nia** [comment, marked as answer]
Nie chce wchodzic w klimat "patrz dokumntacje", ale jednak w tym wypadku zapytam czy ogladales ten link bo tutaj jest to dosc dobrze wyjasnione :) [https://docs.microsoft.com/en-us/azure/logic-apps/logic-apps-pricing](https://docs.microsoft.com/en-us/azure/logic-apps/logic-apps-pricing)

**Reply by jakubigla** [reply]
Tak, czytalem oczywiscie ale nadal nie bylo to dla mnie jasne (nie odwazylbym sie w tej grupie bez przeczytania czego kolwiek haha).
To co znalazlem gdzie indziej to:
`Built-in connectors: Such as calling HTTP endpoints, calling Azure Functions, and Azure API Management operations as well as data manipulating actions. These actions will count as a normal action execution.`

Czyli place po prostu za akcje. Standardowy connector to jak chcesz zintegrowac np z service busem (czego aktualnie nie potrzebuje). Wyglada na to ze jednak koszty sa takie same.

**Reply by wi5nia** [reply]
Tak, bo genelanie takie skladowe Ci sie na finalny rachunek skladaja
![image](https://user-images.githubusercontent.com/12593852/93444247-cfb73d80-f8d0-11ea-84c6-3ab4840605fd.png)
Wiec patrzysz co jest co w docsach i liczysz 😃

**Reply by jakubigla** [reply]
Zgadza sie, bardziej mi chodzilo o to, pod jaka kategorie podchodzi wywolanie azure function :) wyglada na to ze w tym kalkulatorze dla mojego use case'u to same akcje do obliczenia.

**Reply by wi5nia** [reply]
[https://docs.microsoft.com/en-us/azure/connectors/apis-list#enterprise-connectors](https://docs.microsoft.com/en-us/azure/connectors/apis-list#enterprise-connectors)

> Connectors are classified as either Standard or Enterprise. Enterprise connectors provide access to enterprise systems such as SAP, IBM MQ, and IBM 3270 for an additional cost. To determine whether a connector is Standard or Enterprise, see the technical details in each connector's reference page under Connectors overview.

A Azure Functions to sa standard :)

**Reply by jakubigla** [reply]
No chyba jednak nie :D 

https://docs.microsoft.com/en-us/azure/connectors/apis-list#run-code-from-logic-apps

Wg tego azure function to built-in action a nie konektor.

**Reply by jakubigla** [reply]
"Polling" storage account zeby sprawdzic czy sa nowe bloby, to chyba jest konektor, ktory bede musial rozwazyc.

**Reply by wi5nia** [reply]
No tak, troche uproscilem swoje rozumowanie, bardziej chodzilo o fakt, ze NIE sa ent. ;)

**Reply by eklime** [reply]
> "Polling" storage account zeby sprawdzic czy sa nowe bloby, to chyba jest konektor, ktory bede musial rozwazyc.

Zamiast sprawdzania czy coś się pojawiło, lepiej ustawić Event Grid i wywołanie LA dopiero jak się coś pojawi

**Reply by jakubigla** [reply]
good point, w porownywalnej architekturze (ktora tez tworze) dla AWS, mam podobny mechanizm (cloud watch events). "Polling" to jest slaba opcja generalnie.

**Reply by MateuszNad** [reply]
Pewnie podstawy ale dodam, że w kodzie Logic App najlepiej podejrzeć typ akcji. Jeśli jest ApiConnection to jest to connector.
Na końcu zostawie jeszcze coś takiego (jakby co to teraz działa :) - https://csb971ecce36ff2x4b42x9ea.z6.web.core.windows.net/
